python/lsst/daf/butler/core/dimensions/dataId.py

```diff
@@ -0,0 +1,351 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+__all__ = ("DataId",)
+
+import itertools
+from collections.abc import Mapping
+from .graph import DimensionGraph
+from .elements import Dimension
+
+
+class DataId(Mapping):
+    r"""A dict-like identifier for data usable across multiple collections
+    and `DatasetType`\s.
+
+    They keys of a `DataId` correspond to the "link" columns of one or more
```

Comment:
Please move all this content to the [Notes](
https://developer.lsst.io/python/numpydoc.html#notes) section.